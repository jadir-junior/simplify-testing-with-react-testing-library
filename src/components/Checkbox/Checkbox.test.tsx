import { render, screen } from "@testing-library/react";

import { Checkbox } from ".";
import React from "react";

describe("<Checkbox />", () => {
  it("Should render", () => {
    const { container } = render(<Checkbox />);

    expect(container.firstChild).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render a checkbox with label", () => {
    render(<Checkbox label="Remember me" />);

    expect(screen.getByLabelText(/remember me/i)).toBeInTheDocument();
  });

  it("should render a checkbox without label", () => {
    render(<Checkbox />);

    expect(screen.queryByLabelText(/remember me/i)).not.toBeInTheDocument();
  });
});
