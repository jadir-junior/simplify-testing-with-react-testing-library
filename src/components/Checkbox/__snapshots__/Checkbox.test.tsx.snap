// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> Should render 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 8px 0;
}

<div
  class="c0"
>
  <input
    class=""
    type="checkbox"
  />
</div>
`;
